t/007-find-calls.t.cpp

```diff
@@ -0,0 +1,108 @@
+#include "clangmetatool-testconfig.h"
+
+#include <gtest/gtest.h>
+
+#include <clangmetatool/meta_tool_factory.h>
+#include <clangmetatool/meta_tool.h>
+#include <clangmetatool/collectors/find_calls.h>
+
+#include <clang/Frontend/FrontendAction.h>
+#include <clang/Tooling/Core/Replacement.h>
+#include <clang/Tooling/CommonOptionsParser.h>
+#include <clang/Tooling/Tooling.h>
+#include <clang/Tooling/Refactoring.h>
+#include <llvm/Support/CommandLine.h>
+
+class MyTool {
+private:
+  std::string n = std::string("bar");
+  unsigned int a = 0;
+  clang::CompilerInstance* ci;
+  clangmetatool::collectors::FindCalls fc;
+public:
+
+  MyTool(clang::CompilerInstance* ci, clang::ast_matchers::MatchFinder *f)
+    :ci(ci), fc(ci, f, n, a) {}
+
+  const clang::DeclRefExpr* get_pointer_to(const clang::Expr* expr) {
+    const clang::DeclRefExpr* ret = NULL;
+    const clang::UnaryOperator* op =
+      llvm::dyn_cast_or_null<const clang::UnaryOperator>(expr);
+    if (op != NULL) {
+      if (op->getOpcode() == clang::UnaryOperatorKind::UO_AddrOf) {
+        ret = llvm::dyn_cast_or_null<const clang::DeclRefExpr>
+          (op->getSubExpr());
+      }
+    }
+    return ret;
+  }
+
+  void postProcessing
+    (std::map<std::string, clang::tooling::Replacements> &replacementsMap) {
+      clang::SourceManager& sm = ci->getSourceManager();
+
+      std::multimap<
+                    const clang::FunctionDecl*,
+                    const clang::CallExpr*> *call_context = &(fc.getData()->call_context);
```

Comment:
You can use `auto` to avoid this excessive typing... Also, use references instead of pointers...

Reply:
this seems to be the case in other tests: https://github.com/bloomberg/clangmetatool/blob/master/t/004-variablerefs-basic.t.cpp#L31